### PR TITLE
Add support for randomly writing ooo samples

### PR DESCRIPTION
**Pull request by codesome**

### CODE REVIEW

File: cmd/generator.go [hunk 1]
```diff
@@ -27,6 +29,8 @@ var (
 	queryMaxAge            = kingpin.Flag("query-max-age", "How back in the past metrics can be queried at most.").Default("24h").Duration()
 	tenantsCount           = kingpin.Flag("tenants-count", "Number of tenants to fake.").Default("1").Int()
 	seriesCount            = kingpin.Flag("series-count", "Number of series to generate for each tenant.").Default("1000").Int()
+	oooSeriesCount         = kingpin.Flag("out-of-order-series-count", "Number of out of order series to generate for each tenant.").Default("0").Int()
+	maxOOOTime             = kingpin.Flag("max-out-of-order-time", "How old an out of order sample can be *in minutes*.").Default("40").Int()
```
Comment (codesome) [inline cmd/generator.go:33]: Default to 40m because with 30m of OOO Allowance, we will be able to test if limits are working properly or not

**Comment by codesome** [comment]
@pracucci it would be great to get a sanity check on this from you. (we wont be blocked, so there is no pressure)

**Review by jesusvazquez — APPROVED** [review]
LGTM

I was thinking that maybe the name of the metrics like `cortex_load_generator_out_of_order_sine_wave` should be a constant but I'm leaving this up to you.

**Comment by codesome** [comment]
Since currently it was not a constant, it did not have it as a constant. Merging since it is a small codebase and not much gain with a constant. We can look at it in the future if we have more timeseries.

**Comment by codesome** [comment]
I just realised that this PR was pointing to the master branch. I manually (1) synced out-of-order branch with master to get this change (2) force pushed the master to remove this commit.

**Pull request merged**